### PR TITLE
fix: use modifiedIndex as lru key when merge plugins from route and consumer

### PR DESCRIPTION
**Pull request by tzssangglass**
…consumer

### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugin.lua [hunk 1]
```diff
@@ -655,7 +655,7 @@ function _M.merge_consumer_route(route_conf, consumer_conf, api_ctx)
     core.log.info("route conf: ", core.json.delay_encode(route_conf))
     core.log.info("consumer conf: ", core.json.delay_encode(consumer_conf))
 
-    local flag = tostring(route_conf) .. tostring(consumer_conf)
+    local flag = route_conf.modifiedIndex .. consumer_conf.modifiedIndex
```
Comment (spacewander) [inline apisix/plugin.lua:658]: We should use both id and the version (modifiedIndex)

Comment (tzssangglass) [reply apisix/plugin.lua]: done

File: apisix/plugin.lua [hunk 2]
```diff
@@ -655,7 +655,8 @@ function _M.merge_consumer_route(route_conf, consumer_conf, api_ctx)
     core.log.info("route conf: ", core.json.delay_encode(route_conf))
     core.log.info("consumer conf: ", core.json.delay_encode(consumer_conf))
 
-    local flag = tostring(route_conf) .. tostring(consumer_conf)
+    local flag = route_conf.value.id .. "#" .. route_conf.modifiedIndex
+                 .. consumer_conf.id .. "#" .. consumer_conf.modifiedIndex
```
Comment (spacewander) [inline apisix/plugin.lua:659]: Better to add a `#` between route_conf.modifiedIndex and consumer_conf.id

Comment (tzssangglass) [reply apisix/plugin.lua]: done

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**